subsys/caf/events/ble_common_event.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2018-2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <assert.h>
+#include <sys/util.h>
+#include <stdio.h>
+
+#include <caf/events/ble_common_event.h>
+
+
+static const char * const state_name[] = {
+	"DISCONNECTED",
+	"DISCONNECTING",
+	"CONNECTED",
+	"SECURED",
+	"CONN_FAILED"
+};
```

Comment:
It's sad we can't use X-macro here because of the check patch.. This starts to be really annoying because this may introduce a bug if someone decides to extend this and forgets about the enum in the header file.

Reply:
We got an assert that ensures that at least the ARRAY_SIZE is proper. I think that's all we could do here.